### ISSUE TITLE
:set newtab about:home - Blank New Tab

### ISSUE THREAD

**Issue by t3chman** [tridactyl/tridactyl#1090]
*   Brief description of the problem:

:set newtab [about:*] (e.g. :set newtab about:home) doesn't open the relevant about:* page, like about:home (also seems like any URI with "https://" doesn't work). I tried about:preferences and had the same results. After setting an about:* page or "https://" URI, new tabs are blank.

I like the about:home page, because it auto-selects the address-bar while providing recently visited pages and add-on/extension embedded recommendations.

*   Steps to reproduce:

    1.  :set newtab about:home
    2.  Ctrl-t or open new tab
    3.  New blank tab is opened

*   Tridactyl version (`:version`): 1.14.0-1.14.0

*   Firefox version (Top right menu > Help > About Firefox): 62.0.3 (64-bit)

*   URL of the website the bug happens on: about:* (any about:* page, e.g. about:home)

*   Config (in a new tab, run `:viewconfig`, copy the url and paste it somewhere like gist.github.com): https://gist.github.com/t3chman/49c614f7d479491325077554b2dd052a

*   Contents of ~/.tridactylrc or ~/.config/tridactyl/tridactylrc (if they exist): running linux, but they do not exist

**Comment by antonva** [comment]
Not a proper solution but some things to consider:

You can set the newtab page to `about:home` if you use the native messenger but it will annoyingly open two tabs, one being blank, the other being `about:home`. (It will focus on `about:home` though)

As to having the url/omnibar focused on tridactyl's newtab page, you can `set newtabfocus urlbar` to restore that functionality.

**Comment by t3chman** [comment]
Ahh, thank you! Good temporary solutions!

**Comment by nikitavoloboev** [comment]
Is this still an issue. If I understand correctly, there is no way to have a blank page (ideally black color) with this extension?

**Comment by t3chman** [comment]
Correct, this is still an issue. I'm unable to get a blank page to work or the :set newtabfocus urlbar consistently.

**Comment by glacambre** [comment]
>I'm unable to get a blank page

`:set newtab about:blank` works for me, did you try that?

>or the :set newtabfocus urlbar consistently

Very surprising, the URLbar is selected by default and we're only deselecting it the setting is `page`. Are you running the betas? Does this stop happening when Tridactyl is disabled?

**Comment by nikitavoloboev** [comment]
> `:set newtab about:blank` works for me, did you try that?

This works for me. Thank you!

Wish the page was dark color though. With Mojawe, it's really striking to have a full white page.

**Comment by glacambre** [comment]
If you have the native messenger installed, create `$TRIDACTYL_CONFIG_FOLDER/themes/black.css`, put
```css
html.TridactylOwnNamespace {
    background: black;
}
```
in this file, then in Tridactyl do `colourscheme black`, this should turn your newtab page black. See `:help colourscheme` for more info.

**Comment by nikitavoloboev** [comment]
> If you have the native messenger installed

I just googled what `native messenger` is but found no link to download of it.

**Comment by glacambre** [comment]
Oh, sorry, it's an external application that Tridactyl can use in order to communicate with your OS: use your favorite text editor to edit Firefox's text inputs, read a config file from your disk, that kind of thing. You can run `:nativeinstall` in order to get installation instructions. If you want to take a look at its source code beforehand it's available here: https://github.com/tridactyl/tridactyl/blob/master/native/native_main.py .

If you don't want to install it, you can try running `:set customthemes.black html.TridactylOwnNamespace{background: black;}` and `set theme black` but I'm not sure it'll work.

**Comment by nikitavoloboev** [comment]
> You can run `:nativeinstall` in order to get installation instructions

Got this running the command: 

![2018-10-18 at 23 09](https://user-images.githubusercontent.com/6391776/47184447-f9553c00-d32a-11e8-9aa8-0d7cf4067e1d.png)




> If you don't want to install it, you can try running

Tried running that, nothing happened. I hope I didn't break anything.

**Comment by glacambre** [comment]
Ah, that's because the command is actually `:installnative`. Sorry.

**Comment by bovine3dom** [comment]
`nativeinstall` does fit better with the rest of our commands. Perhaps we should rename it and leave an alias.

**Comment by t3chman** [comment]
> > I'm unable to get a blank page
> 
> `:set newtab about:blank` works for me, did you try that?
> 
> > or the :set newtabfocus urlbar consistently
> 
> Very surprising, the URLbar is selected by default and we're only deselecting it the setting is `page`. Are you running the betas? Does this stop happening when Tridactyl is disabled?

Not running a beta, as I installed the latest stable from the Firefox marketplace. The focus is always on the URL bar without Tridactyl running. But when setting any custom newtab with Tridactyl, the focus is mostly on the page and not the URL bar, although on a few occasions I did see it focus on the URL bar, but not consistently.

I like the idea of the add-on, but I've decided not to run it while all browsing and interaction data is run through it. I noticed a performance hit and with the bugs I encountered, it doesn't seem stable enough to use day-to-day. I appreciate the work and support though! I hope to use it in the future.
